src/ResourceManager/Sql/Commands.Sql/ManagedDatabase/Cmdlet/GetAzureSqlManagedDatabase.cs

```diff
@@ -0,0 +1,129 @@
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Sql.ManagedDatabase.Model;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.ManagedDatabase.Cmdlet
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmSqlManagedDatabase", ConfirmImpact = ConfirmImpact.None)]
+    public class GetAzureSqlManagedDatabase : AzureSqlManagedDatabaseCmdletBase<IEnumerable<AzureSqlManagedDatabaseModel>>
+    {
+        protected const string GetByNameAndResourceGroupParameterSet =
+            "GetManagedDatabaseFromInputParameters";
+
+        protected const string GetByResourceIdParameterSet =
+            "GetManagedDatabaseFromAzureResourceId";
+
+        /// <summary>
+        /// Gets or sets the name of the managed database to use.
+        /// </summary>
+        [Parameter(ParameterSetName = GetByNameAndResourceGroupParameterSet,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "The name of the Azure SQL Managed Database to retrieve.")]
+        [Alias("Name")]
+        [ValidateNotNullOrEmpty]
+        public string ManagedDatabaseName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the Azure Sql Managed Instance
+        /// </summary>
+        [Parameter(ParameterSetName = GetByNameAndResourceGroupParameterSet,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = "The Azure Sql Managed Instance name.")]
+        [ValidateNotNullOrEmpty]
+        public override string ManagedInstanceName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the resource group to use.
+        /// </summary>
+        [Parameter(ParameterSetName = GetByNameAndResourceGroupParameterSet,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2,
+            HelpMessage = "The name of the resource group.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public override string ResourceGroupName { get; set; }
```

Comment:
Generally, we have positions going from top level resource down.  So here, you should have Get-AzureRmSqlManagedDatabase -ResourceGroupName -ManagedInstanceName -ManagedDatabaseName

Reply:
I used the pattern how it is done for Get-AzureRmSqlDatabase, and there we have DatabaseName first, then ServerName and at the end ResourceGroupName